snapcraft/plugins/_python/_pip.py

```diff
@@ -0,0 +1,339 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016-2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import collections
+import contextlib
+import json
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import snapcraft
+from snapcraft import file_utils
+from ._python_finder import (
+    get_python_command,
+    get_python_headers,
+    get_python_home,
+)
+from . import errors
+
+logger = logging.getLogger(__name__)
+
+
+def _process_common_args(args, *, packages, constraints,
```

Comment:
I like this, but even though it is a list, it would be better to return `args` as well.

Reply:
With the switch in arg ordering, it doesn't need to be passed args at all, which is a nice simplification.

Reply:
Done.